app/src/server.js

```diff
@@ -6,45 +6,50 @@
  * Main entry point for the Digital Citizenship proxy.
  */
 
-require("dotenv").load();
-
+import tokenStrategy from "./strategies/tokenStrategy";
+import spidStrategy from "./strategies/spidStrategy";
 import container from "./container";
-import type { SessionStorageInterface } from "./services/sessionStorageInterface";
 import ProfileController from "./controllers/profileController";
 import AuthenticationController from "./controllers/authenticationController";
 
+require("dotenv").load();
+
+const bodyParser = require("body-parser");
+const fs = require("fs");
 const express = require("express");
+const https = require("https");
 const morgan = require("morgan");
 const passport = require("passport");
-const Strategy = require("passport-http-bearer");
-
-const port = process.env.PORT || 8080;
-
-passport.use(
-  new Strategy(function(token, done) {
-    const sessionStorage = (container.resolve(
-      "sessionStorage"
-    ): SessionStorageInterface);
-    const user = sessionStorage.get(token);
-    if (user) {
-      return done(null, user);
-    } else {
-      return done(null, false);
-    }
-  })
-);
+
+const port = process.env.PORT || 443;
+
+// Setup Passport.
+
+passport.use(tokenStrategy);
+passport.use(spidStrategy);
+
+// Create and setup the Express app.
 
 const app = express();
 app.use(morgan("dev"));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
 app.use(express.static("public"));
 app.use(passport.initialize());
 
-app.get("/sso", function(req: express$Request, res: express$Response) {
+// Setup routing.
+
+app.get("/login", passport.authenticate("spid", { session: false }));
+
+app.post("/acs", passport.authenticate("spid", { session: false }), function(
+  req: express$Request,
+  res: express$Response
+) {
   const controller = (container.resolve(
```

Comment:
shouldn't this be outside the endpoint cb? it's fair to assume the controller will not change from one request to another...

Reply:
if you resolve the controller once, you can then:

``` js
const acsController = ...
app.post("/acs", spidAth, acsController);
```